src/presenter/general-presenter.js

```diff
@@ -3,27 +3,62 @@ import SortView from '../view/sort-view.js';
 import EventListView from '../view/event-list-view.js';
-import {RenderPosition, render} from '../render.js';
+import {RenderPosition} from '../render.js';
+import { render, replace} from '../framework/render.js';
 export default class GeneralPresenter {
-  eventListComponent = new EventListView();
+  #tripControlsFilters = null;
+  #tripEvents = null;
+  #pointModel = null;
+  #tripEventsView = null;
+  #eventListComponent = null;
   constructor({tripControlsFilters,tripEvents,pointModel}) {
-    this.tripControlsFilters = tripControlsFilters;
-    this.tripEvents = tripEvents;
-    this.pointModel = pointModel;
+    this.#tripControlsFilters = tripControlsFilters;
+    this.#tripEvents = tripEvents;
+    this.#pointModel = pointModel;
+    this.#eventListComponent = new EventListView();
   }
 
   init() {
-    const events = [...this.pointModel.getEvents()];
-    const destinations = [...this.pointModel.getDestinations()];
-    const offers = [...this.pointModel.getOffers()];
-    render(new FiltersView(),this.tripControlsFilters,RenderPosition.BEFOREEND);
-    render(new SortView(),this.tripEvents);
-    render(this.eventListComponent,this.tripEvents);
-    // render(new EditingForm({destinations, offers}),this.eventListComponent.getElement());
-    render(new EditingForm({event,destinations, offers}),this.eventListComponent.getElement());
-    events.forEach((event) => {
-      render(new WayPoint({event,destinations, offers}),this.eventListComponent.getElement());
-    });
+
+    render(new FiltersView(),this.#tripControlsFilters,RenderPosition.BEFOREEND);
+    render(new SortView(),this.#tripEvents);
+    this.#renderTripEvents(this.#pointModel);
+    // this.#renderTripEvent();
+    // render(new EditingForm({event,destinations, offers}),this.#eventListComponent.element);
+    // render(tripEventView,this.#eventListComponent.element);
+
+  }
+
+  #renderTripEvents({events}) {
+    render(this.#eventListComponent,this.#tripEvents);
+    events.forEach((event) => this.#renderTripEvent(event));
+  }
+
+  #renderTripEvent(event) {
+    const destinations = [...this.#pointModel.destinations];
+    const offers = [...this.#pointModel.offers];
+    const onEditBtnClick = () => switchToEditMode();
+    const onCloseBtnClick = () => switchToViewMode();
+    const onDocumentKeyDown = (evt) => {
+      if(evt.key === 'Escape') {
+        evt.preventDefault();
+        switchToViewMode();
+      }
+    };
+    const tripEventView = new WayPoint({event,destinations, offers, onClick: onEditBtnClick});
+
+    const eventEditView = new EditingForm({event,destinations, offers, onSubmit: onCloseBtnClick, onClick:onCloseBtnClick});
+    function switchToEditMode() {
+      replace(eventEditView,tripEventView);
+      document.addEventListener('keydown', onDocumentKeyDown);
+
+    }
+    function switchToViewMode() {
+
+      replace(tripEventView,eventEditView);
+    }
+    render(tripEventView, this.#eventListComponent.element);
```

Comment:
Нужно следить за форматированием, это хорошая привычка, чтобы код было легче читать и поддерживать.
1️⃣

```suggestion
    const eventEditView = new EditingForm({event, destinations, offers, onSubmit: onCloseBtnClick, onClick: onCloseBtnClick});

    function switchToEditMode() {
      replace(eventEditView,tripEventView);
      document.addEventListener('keydown', onDocumentKeyDown);
    }

    function switchToViewMode() {
      replace(tripEventView,eventEditView);
    }

    render(tripEventView, this.#eventListComponent.element);
```